back-end/src/main/java/project/Layer.java

```diff
@@ -12,10 +14,13 @@
     private boolean deltaBox;
     private Placement placement;
     private String rendering;
-    private Indexer indexer;
-    private boolean isAutoDDLayer;
+    @Exclude private Indexer indexer;
     private String autoDDId;
-    private boolean isPredicatedTable;
+
+    @SerializedName(
+            value = "indexerClass",
+            alternate = {"indexerType", "indexerName"})
+    private String indexerType;
 
```

Comment:
can you help add a few comments here, describing what @serializedName is for, what the arguments are?

Reply:
Of course

Reply:
thanks for the comments, I now know what serializedName does. However, I wonder why not directly name "indexerType" as "indexerName" in the compiler?

Reply:
Oh, it's just that I personally think that `indexerType` is clearer than `indexerName`. Either would work fine.

Reply:
yeah both seems fine to me. As long as you take one and make them consistent, @serializedName is not needed, right?

Reply:
Yes. I first introduced this annotation to make `indexer` a valid fieldname in the compiler... But since the transient keyword would cause other problems, the meaning of `serializedName` is little... I can just get rid of it.

Reply:
yes just make the name consistent and remove @seralizedName

Reply:
Sure.